pkg/controller/jobframework/reconciler.go

```diff
@@ -761,7 +761,13 @@ func (r *JobReconciler) stopJob(ctx context.Context, job GenericJob, wl *kueue.W
 	}
 
 	if jws, implements := job.(ComposableJob); implements {
-		stoppedNow, err := jws.Stop(ctx, r.client, info, stopReason, eventMsg)
+		reason := stopReason
```

Comment:
In my opinion, ignoring the received reason just for ComposableJob when Evicted condition is hard to follow, 
the caller `stopJob` should provide the right reason for all the jobs.

Reply:
I agree, but right now we don't have a condition in every job before suspending.

We could work on that as a follow up. For now, the pain point is in the pod integration, given that they need to be replaced and it's not now immediately obvious when a pod was deleted because of an Eviction (because the Workload condition can change if it's immediately re-admitted).